treeherder/log_parser/intermittents.py

```diff
@@ -0,0 +1,134 @@
+from treeherder.model.models import Group, GroupStatus, Job, Push
+
+
+def check_and_mark_intermittent(job_id):
+    current_job = Job.objects.get(id=job_id)
+    jtname = current_job.job_type.name.strip("-cf")
+    ids = [current_job.push.id]
+
+    try:
+        _ = int(jtname.split("-")[-1])
+        jtname = "-".join(jtname.split("-")[:-1])
+    except ValueError:
+        pass
+
+    # if we are not on try, look at recent history
```

Comment:
This logic will miss if multiple branches are used, e.g. on the release branch for the current (X.0.a) and the next release (X+1.0).

Reply:
for try we only want to look at a single push, every other branch we want to go back in time and look at recent pushes.  Autoland needs 20 pushes in order to find the last full set of tests.